src/hwloc/config/hwloc.m4

```diff
@@ -1179,15 +1176,21 @@ EOF])
 
     AS_IF([test "$hwloc_mode" = "embedded"],
           [HWLOC_EMBEDDED_CFLAGS=$HWLOC_CFLAGS
+           HWLOC_EMBEDDED_VISIBILITY_CFLAGS=$HWLOC_VISIBILITY_CFLAGS
```

Comment:
What are embedded visibility flags? Embedding in my mind means static linking. Visibility only applies to shared libraries.

Reply:
On second thought, this sounds familiar to an issue I ran into with MPL and visibility where I had to do something similar.

Reply:
The visibility flags are included in the MPICH build, and will cause MPICH to not build correctly in some cases.  The above patch allows us to distinguish those cases and disable them in those cases.